platform/src/main/java/org/hillview/table/columns/CategoryEncoding.java

```diff
@@ -28,30 +31,52 @@
  */
 public class CategoryEncoding {
     // Map categorical value to a small integer
-    private final Object2IntOpenHashMap<String> encoding;
+    private final Object2IntOpenHashMap<String> intEncoding;
     // Decode small integer into categorical value
-    private final Int2ObjectOpenHashMap<String> decoding;
+    private final Int2ObjectOpenHashMap<String> intDecoding;
+    // Map categorical value to a byte
+    private final Object2ByteOpenHashMap<String> byteEncoding;
```

Comment:
I don't think you need to change this class. There will be no significant savings by doing this.